src/api/model/modelApi.ts

```diff
@@ -14,57 +14,128 @@
  * under the License.
  */
 
-import { TransactionApi } from '../transaction/transactionApi';
-import {
-  InstallAction,
-  ListSourceAction,
-  Model,
-  ModifyWorkspaceAction,
-} from '../transaction/types';
-
-export class ModelApi extends TransactionApi {
+import _ from 'lodash';
+
+import { ExecAsyncApi } from '../query/execAsyncApi';
+import { QueryInput } from '../query/types';
+import { Model } from '../transaction/types';
+export class ModelApi extends ExecAsyncApi {
   async installModels(database: string, engine: string, models: Model[]) {
-    const action: InstallAction = {
-      type: 'InstallAction',
-      sources: models,
-    };
+    const randInt = Math.floor(Math.random() * Number.MAX_SAFE_INTEGER);
+    const queries: string[] = [];
+    const queryInputs: QueryInput[] = [];
+
+    models.map(model => {
+      const inputName = `${model.name}_input_${randInt}`;
+      queryInputs.push({ name: inputName, value: model.value });
+      queries.push(`def delete:rel:catalog:model["${model.name}"] = rel:catalog:model["${model.name}"]
+        def insert:rel:catalog:model["${model.name}"] = ${inputName}`);
+    });
 
-    return await this.runActions(database, engine, [action], false);
+    return await this.exec(
+      database,
+      engine,
+      queries.join('\n'),
+      queryInputs,
+      false,
+    );
+  }
+
+  async installModelsAsync(database: string, engine: string, models: Model[]) {
+    const randInt = Math.floor(Math.random() * Number.MAX_SAFE_INTEGER);
+    const queries: string[] = [];
+    const queryInputs: QueryInput[] = [];
+
+    models.map(model => {
+      const inputName = `${model.name}_input_${randInt}`;
+      queryInputs.push({ name: inputName, value: model.value });
+      queries.push(`def delete:rel:catalog:model["${model.name}"] = rel:catalog:model["${model.name}"]
+        def insert:rel:catalog:model["${model.name}"] = ${inputName}`);
+    });
+
+    return await this.execAsync(
+      database,
+      engine,
+      queries.join('\n'),
+      queryInputs,
+      false,
+    );
   }
 
   async listModels(database: string, engine: string) {
-    const action: ListSourceAction = {
-      type: 'ListSourceAction',
-    };
+    const randInt = Math.floor(Math.random() * Number.MAX_SAFE_INTEGER);
+    const outName = `models${randInt}`;
+    const rsp = await this.exec(
+      database,
+      engine,
+      `def output:${outName}[name] = rel:catalog:model(name, _)`,
+    );
 
-    return await this.runActions(database, engine, [action]);
+    const result = rsp.results.find(r => {
+      const val =
+        r.metadata.arguments[1].constantType?.value?.arguments[0].value;
+      if (val?.oneofKind === 'stringVal') {
+        const stringVal = new TextDecoder().decode(val.stringVal);
+        if (stringVal.includes(outName)) {
```

Comment:
Why `.includes` instead of the strict comparison like `return stringVal === outName`?